### DISCUSSION TITLE
Display features

### DISCUSSION THREAD

**Discussion by tombloemker** [toverainc/willow#213] [Ideas]
How about an option for the display to show a clock instead of going dark?  
Two options black font with white background or white font with black background.  
Is it possible to do this with a much larger font?

**Comment by kristiankielhofner** [comment]
The hardware and display libraries we use can definitely render a clock. We have NTP support in Willow for this exact use case (plus logging, of course).

The challenge is being methodical and deliberate in terms of user interface and experience. Do you want a clock? What kind? Should it (via LCD backlight control) be on all of the time? The answers to those questions may seem fairly straightforward but people get very passionate when it comes to design decisions like this and we want to make sure we implement such functionality in a way that is forward looking and acceptable to as many users as possible.

Short answer is, we definitely have plans for it but it will likely be a while (well after our first stable release) for us to consider all of these factors and implement it.

**Reply by tombloemker** [reply]
I assumed as much, the user experience can be a little underwhelming as merely a smart home device controller. Amazon's echo will be a problem as far as managing expectations for willow. Maybe this discussion can serve as landing place for display options? It would be nice to know upfront what a clock would cost in resources, are there limitations?  Would the resolution of the display even support a quality looking non-digital clock face with hands? Since there is a touch screen, can clock customization settings be handled within the unit? It would make color selection far more manageable.           
![clock](https://github.com/toverainc/willow/assets/93683913/7e19487d-a0ff-47f1-ac09-74a75c8584ee)

**Reply by kristiankielhofner** [reply]
This is the exact opposite of what we're hearing from users in terms of user experience feedback.

Willow is a voice interface. It's primary purpose is to sit there until you want to talk to it, do whatever you say by sending it where you want, tell you what happened, and then go back to sleep. There has been a tremendous amount of blowback and criticism for Echo devices because they use their displays and increasingly aggressive follow-up prompts to largely serve Amazon's primary interest with these devices: getting you to buy more stuff from Amazon. When you buy an Echo with a display you are installing an Amazon billboard in your environment. As their Echo division continues to bleed more and more money this is only getting worse. Every time I walk by one of my Echo devices I wish I could just turn the display off and it seems most Willow users are in agreement.

The primary user experience expectation for Willow is that it does its job as a voice interface and gets out of the way until you need it again. Echo devices, with their always on displays, are obnoxious and this is exactly what people appreciate about Willow - it's the furthest thing from obnoxious.

Point being, in many ways our targeted user experience is "the opposite of what Echo does". That said we're planning a clock because it's actually useful while not being obnoxious and intrusive. If you want an idea of what a clock could look like search for "LVGL clock".

I should also point out from concept to now Willow is three months old. Echo is eight years old. Other open source alternatives like Rhasspy are four years old. This project is in its infancy and display-related features are towards the bottom of our priority list at the moment.

**Comment by janstadt** [comment]
Any movement here on the clock idea? I Had a nice clock interface while running esphome on my s3 devices that pulls time from my HA instance and would love that as well. I have one next to my bed that i use for the clock and one mounted via a usb port in a wall socket on our kitchen backsplash and having a clock always displaying is nice.
